src/Illuminate/Database/DetectsConcurrencyErrors.php

```diff
@@ -16,6 +36,13 @@ trait DetectsConcurrencyErrors
      */
     protected function causedByConcurrencyError(Throwable $e)
     {
+        if (
+            is_callable($this->concurrencyErrorCheck)
```

Comment:
logic seems wrong. if is callable but returns false, it's output is ignored and the original is used

Reply:
Not entirely sure what you mean. But I'll answer to what I think you mean.
My goal was to get a way to add more definitions of what a (in this case) concurrency error looks like. If we just returned the result from `concurrencyErrorCheck()`. It wouldn't do the existing checks. And I would have to reimplement the existing checks.
However I do agree it seems strange to only use it if it's true. But I'm not sure how the logic should work another way.